targets/TARGET_Ambiq_Micro/TARGET_Apollo3/device/serial_api.c

```diff
@@ -0,0 +1,451 @@
+  };
+
+  do {
+    am_hal_uart_transfer(obj->serial.uart_control->handle, &am_hal_uart_xfer_write_single);
+  } while (bytes_sent == 0);
+}
+
+int serial_readable(serial_t *obj) {
+  MBED_ASSERT(obj->serial.uart_control != NULL);
+  return !(UARTn(obj->serial.uart_control->inst)->FR_b.RXFE);
+}
+
+int serial_writable(serial_t *obj) {
+  MBED_ASSERT(obj->serial.uart_control != NULL);
+  return !(UARTn(obj->serial.uart_control->inst)->FR_b.TXFF);
+}
+
+void serial_clear(serial_t *obj) {
+  // todo:
+  MBED_ASSERT(0);
+}
+
+void serial_break_set(serial_t *obj) {
+  MBED_ASSERT(obj->serial.uart_control != NULL);
+  UARTn(obj->serial.uart_control->inst)->LCRH |= UART0_LCRH_BRK_Msk;
+}
+
+void serial_break_clear(serial_t *obj) {
+  MBED_ASSERT(obj->serial.uart_control != NULL);
+  UARTn(obj->serial.uart_control->inst)->LCRH &= ~UART0_LCRH_BRK_Msk;
+}
+
+void serial_pinout_tx(PinName tx) {
+  // todo: vestigial?
+  MBED_ASSERT(0);
+}
+
+#if DEVICE_SERIAL_FC
+
+void serial_set_flow_control(serial_t *obj, FlowControl type, PinName rxflow, PinName txflow) {
+  // todo:
+  MBED_ASSERT(0);
+}
+
+void serial_set_flow_control_direct(serial_t *obj, FlowControl type, const serial_fc_pinmap_t *pinmap) {
+  // todo:
+  MBED_ASSERT(0);
+}
+#endif
+
+const PinMap *serial_tx_pinmap(void) {
+  return PinMap_UART_TX;
+}
+
+const PinMap *serial_rx_pinmap(void) {
+  return PinMap_UART_RX;
+}
+
+#if DEVICE_SERIAL_FC
+
+const PinMap *serial_cts_pinmap(void) {
+  return PinMap_UART_CTS;
+}
+
+const PinMap *serial_rts_pinmap(void) {
+  return PinMap_UART_RTS;
+}
+#endif
+
+#if DEVICE_SERIAL_ASYNCH
+
+int serial_tx_asynch(serial_t *obj, const void *tx, size_t tx_length, uint8_t tx_width, uint32_t handler, uint32_t event, DMAUsage hint)
+{
+  MBED_ASSERT(obj->serial.uart_control != NULL);
+  uint32_t bytes_written = 0;
+
+  am_hal_uart_transfer_t am_hal_uart_xfer_write = {
+    .ui32Direction = AM_HAL_UART_WRITE,
+    .pui8Data = (uint8_t *)obj->tx_buff.buffer,
+    .ui32NumBytes = tx_length, // todo: consider maybe this? (uint32_t)obj->tx_buff.length,
+    .ui32TimeoutMs = 0,
+    .pui32BytesTransferred = &bytes_written,
+  };
+
+  am_hal_uart_transfer(obj->serial.uart_control->handle, &am_hal_uart_xfer_write);
+
+  return (int)bytes_written;
+}
+
+void serial_rx_asynch(serial_t *obj, void *rx, size_t rx_length, uint8_t rx_width, uint32_t handler, uint32_t event, uint8_t char_match, DMAUsage hint)
+{
+  // todo: revisit
+  MBED_ASSERT(obj->serial.uart_control != NULL);
+  uint32_t bytes_read = 0;
+
+  am_hal_uart_transfer_t am_hal_uart_xfer_read = {
+    .ui32Direction = AM_HAL_UART_READ,
+    .pui8Data = (uint8_t *)obj->rx_buff.buffer,
+    .ui32NumBytes = rx_length, // todo: consider this (uint32_t)obj->rx_buff.length,
+    .ui32TimeoutMs = 0,
+    .pui32BytesTransferred = &bytes_read,
+  };
+
+  am_hal_uart_transfer(obj->serial.uart_control->handle, &am_hal_uart_xfer_read);
+}
+
+uint8_t serial_tx_active(serial_t *obj) {
+  // todo:
+  MBED_ASSERT(0);
```

Comment:
if async is not supported, lets remove these functions

Reply:
removed